openwisp_users/models.py

```diff
@@ -59,6 +63,19 @@ def organizations_pk(self):
                     .values_list('organization_id')
         return qs
 
+    def clean(self):
+        """
+        If a user has an email set, validate the unique constraint
+        else don't validate this constraint.
+        If no email is set, raise a validation error
+        """
+        if self.email:
+            if User.objects.exclude(pk=self.pk).filter(email=self.email).exists():
+                raise ValidationError({'email': 'A user with this email already exist'})
+        else:
+            raise ValidationError({'email': 'Please this field is required'})
```

Comment:
we cannot do this or modules that need to create users without email won't work, this can happen sometimes, read here: https://django-freeradius.readthedocs.io/en/latest/general/generating_users.html

This is one of those few times in which we need to use the form level validation instead of the model level validation: we allow the email to be blank on the model, so we leave that option open for any feature that may need it, but we use the form level validation in the admin to enforce the email to be present when admins use the web interface to add users